repose-aggregator/components/filters/header-normalization-filter/src/main/scala/org/openrepose/filters/headernormalization/HeaderNormalizationFilter.scala

```diff
@@ -71,22 +73,42 @@ class HeaderNormalizationFilter @Inject()(configurationService: ConfigurationSer
 
     val wrappedRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
 
-    config find { target =>
+    configRequest find { target =>
       // find the first "target" config element that matches this request (if any)
       target.url.matcher(wrappedRequest.getRequestURI).matches &&
         (target.methods.contains(wrappedRequest.getMethod) || target.methods.contains(AllHttpMethods))
     } foreach { target =>
       // figure out which headers to remove, and remove them
       (target.access match {
         case WhiteList => wrappedRequest.getHeaderNamesScala.diff(target.headers)
-          // written like this to maintain the case-insensitive string comparisons - do not swap
-        case BlackList => target.headers.intersect(wrappedRequest.getHeaderNamesScala)
+        case BlackList => target.headers
       }).foreach(wrappedRequest.removeHeader)
 
-      metricsMeter.mark(s"${target.url}_${wrappedRequest.getMethod}")
+      metricsMeter.mark(s"${target.url}_${wrappedRequest.getMethod}_request")
     }
 
-    filterChain.doFilter(wrappedRequest, servletResponse)
+    val wrappedResponse = if (configResponse.isEmpty) None else Option(
+      new HttpServletResponseWrapper(servletResponse.asInstanceOf[HttpServletResponse], MUTABLE, PASSTHROUGH))
+
+    filterChain.doFilter(wrappedRequest, wrappedResponse.getOrElse(servletResponse.asInstanceOf[HttpServletResponse]))
+
+    configResponse find { target =>
+      // find the first "target" config element that matches this request (if any)
+      target.url.matcher(wrappedRequest.getRequestURI).matches &&
+        (target.methods.contains(wrappedRequest.getMethod) || target.methods.contains(AllHttpMethods))
+    } foreach { target =>
+      // figure out which headers to remove, and remove them
+      (target.access match {
+        // The Response Wrapper doesn't return a case-insensitive collection of Header Names at this time.
+        case WhiteList => (new TreeSet[String]()(caseInsensitiveOrdering) ++
+          wrappedResponse.get.getHeaderNames.asScala).diff(target.headers)
```

Comment:
So for this bit, I think @adrianjgeorge pointed out that you could just call `.toLowercase` everything before you do the comparison. That is, lowercase the configured header names on configuration updated, and the response header names when you get them. Then the comparison would look something like:
`wrappedResponse.get.getHeaderNames.asScala.filterNot(h => target.headers.contains(h.toLowerCase))`

If you'd rather make use of Ordering to try and keep things clean (and maintain `Set` semantics) you could just make the `Set` you create for the configured header names a `TreeSet` up front. I'll add more comments after this one to show the changes necessary for that approach.